src/Data/Textile/Query.elm

```diff
@@ -361,6 +362,29 @@ updateStepCountry label code query =
             query
 
 
+validateMaterials : List MaterialQuery -> Result String (List MaterialQuery)
+validateMaterials materials =
+    if materials == [] then
+        Ok []
+
+    else
+        let
+            total =
+                materials
+                    |> List.map (.share >> Split.toFloat)
+                    |> List.sum
+        in
+        -- Note: taking care of float number rounding precision errors https://en.wikipedia.org/wiki/Round-off_error
+        if not (List.member total [ 1, 0.6 + 0.3 + 0.1 ]) then
```

Comment:
Instead of having to deal with floats and this kind of JS silliness, why is the `Split` type not using `Int` by default?

```elm
type Split
    = Split Int
```
To my understanding it's modeling shares that can be expressed as an `Int` percentage. So here we should check if the sum is equal to `100`.

It looks like it would solve this problem and converting from an `Int` to a `Float` when really needed should be easier than dealing with rounding stuff.

Reply:
The internal repr can be either int or float, hence float. Also, reverting #525 would reintroduce related bugs. I think it's safe to go the route provided with this patch.

Reply:
Thanks for the precision!